trivor-subtitles/lib/subtitle.js

```diff
@@ -0,0 +1,148 @@
+'use strict'
+
+/**
+ * Module dependencies.
+ */
+
+const MovieDB = require('./themoviedb.js');
+const Queue = require('./queue.js');
+const opensubtitles = require('./opensubtitles.js');
+const request = require('request');
+const zlib = require('zlib');
+const fs = require('fs');
+const config = require('config');
+
+/**
+ * Manages the download and upload of subtitles
+ * 
+ */
+class Subtitle {
+
+    /**
+     * Subtitle constructor.
+     *
+     * @param {String} imdbId
+     * @param {Queue} subtitles queue
+     * @api private
+     */
+    constructor(imdbId, subtitlesQ) {
+        this.imdbId = imdbId;
+        this.mdb = new MovieDB(imdbId);
+        this.subtitlesQ = subtitlesQ;
+        this.subtitles = [];
+    }
+
+    /**
+     * Checks if the provided id is of a movie.
+     *
+     * @api private
+     */
+    isMovie() {
+        return this.title.movie_results.length;
+    }
+
+    /**
+     * Checks if the provided id is of a tv show.
+     *
+     * @api private
+     */
+    isTVShow() {
+        return this.title.tv_results.length;
+    }
+
+    /**
+     * Returns all subtitles of the title
+     *
+     * @api public
+     */
+    getSubtitles() {
+        return this.subtitles;
+    }
+
+    /**
+     * Adds all subtitles of a title.
+     *
+     * @param {Function} callback
+     * 
+     * @api private
+     */
+    addSubtitles(callback) {
```

Comment:
I think we should send other fields too such as `year`, `name` and `type`(TV_SHOW, MOVIE). Is it possible?